### PR TITLE
make user relation polymorphic

### LINKED ISSUE #33: Ability to set multiple models as readers
I have a situation where I need multiple models to act as reader, however this isn't currently possible in unread gem. What would be a workaround?

### PR DESCRIPTION
**Pull request by everplays**
fixes #33.

### CODE REVIEW

**Comment by everplays** [comment]
@ledermann I didn't add new tests as the test coverage is pretty good but let me know if you like to have some explicit testing for polymorphic user.

File: lib/unread/readable.rb [hunk 1]
```diff
@@ -27,8 +27,9 @@ def mark_array_as_read(array, user)
             if global_timestamp && global_timestamp >= timestamp
               # The object is implicitly marked as read, so there is nothing to do
             else
-              rm = obj.read_marks.where(:user_id => user.id).first || obj.read_marks.build
+              rm = obj.read_marks.where(:user_id => user.id, :user_id => user.class.base_class.name).first || obj.read_marks.build
```
Comment (ledermann) [inline lib/unread/readable.rb]: ... this must be `:user_type => `

**Comment by ledermann** [comment]
Thank you @everplays, great PR! Some additional ideas:

* To improve naming things, we should rename the polymorphic database column from `user` to `reader`, so we have `reader_id` and `reader_type`. This naming would better match to `acts_as_reader`
* We should add a migration to support converting existing read_marks
* We should add some tests handling two different reader classes (because they are missing yet, the typo above does not break the tests)

**Comment by everplays** [comment]
@ledermann sure. I can make these changes. However, changing `user` to `reader` will break all existing codes. Are you sure that we should do that?

**Comment by everplays** [comment]
@ledermann I've applied two of your comments (out of four). Let me know if you want to break BC so I can add the migration and change `user` to `reader`. Otherwise, I'll add the migration and this should be good to merge.

**Comment by ledermann** [comment]
Fine!

But as I see it we are already introducing breaking changes, because the database schema requires the new column `user_type`. Without a migration the gem will not work with old data.

In addition, if a developer is using the public API of the gem only, changing some column names in the `read_marks` table will **not** break existing code (except for the needed migration) - or am I wrong?

In a nutshell, I prefer to rename the column names.

**Comment by everplays** [comment]
@ledermann done. I've added some documentation as well. Feel free to remove/alter them.

**Comment by ledermann** [comment]
Great, thank you very much @everplays! Will merge it soon...
